Neos.ContentRepository/Classes/TypeConverter/NodeLikeConverterHelperTrait.php

```diff
@@ -0,0 +1,135 @@
+                    } else {
+                        $nodePropertyValue = null;
+                    }
+                    break;
+                case 'integer':
+                    $nodePropertyValue = intval($nodePropertyValue);
+                    break;
+                case 'boolean':
+                    if (is_string($nodePropertyValue)) {
+                        $nodePropertyValue = $nodePropertyValue === 'true' ? true : false;
+                    }
+                    break;
+                case 'array':
+                    $nodePropertyValue = json_decode($nodePropertyValue, true);
+                    break;
+            }
+            if (substr($nodePropertyName, 0, 1) === '_') {
+                $nodePropertyName = substr($nodePropertyName, 1);
+                ObjectAccess::setProperty($node, $nodePropertyName, $nodePropertyValue);
+                continue;
+            }
+
+            if (!isset($nodeTypeProperties[$nodePropertyName])) {
+                if ($configuration !== null && $configuration->shouldSkipUnknownProperties()) {
+                    continue;
+                } else {
+                    throw new TypeConverterException(sprintf('Node type "%s" does not have a property "%s" according to the schema', $nodeType->getName(), $nodePropertyName), 1359552744);
+                }
+            }
+            $innerType = $nodePropertyType;
+            if ($nodePropertyType !== null) {
+                try {
+                    $parsedType = TypeHandling::parseType($nodePropertyType);
+                    $innerType = $parsedType['elementType'] ?: $parsedType['type'];
+                } catch (InvalidTypeException $exception) {
+                }
+            }
+
+            if (is_string($nodePropertyValue) && $objectManager->isRegistered($innerType) && $nodePropertyValue !== '') {
+                $nodePropertyValue = $propertyMapper->convert(json_decode($nodePropertyValue, true), $nodePropertyType, $configuration);
+            }
+            $node->setProperty($nodePropertyName, $nodePropertyValue);
+        }
+    }
+
+    /**
+     * Prepares the context properties for the nodes based on the given workspace and dimensions
+     *
+     * @param array $source
+     * @param PropertyMappingConfigurationInterface $configuration
+     * @return array
+     */
+    protected function prepareContextProperties(array $source, PropertyMappingConfigurationInterface $configuration = null)
+    {
+        $contextProperties = [
+            'workspaceName' => $source['__workspaceName'] ?? 'live',
+            'invisibleContentShown' => false,
+            'removedContentShown' => false
+        ];
+        if ($contextProperties['workspaceName'] !== 'live') {
+            $contextProperties['invisibleContentShown'] = true;
+            if ($configuration !== null && $configuration->getConfigurationValue(self::class, self::REMOVED_CONTENT_SHOWN) === true) {
```

Comment:
The trait depends on a constant not available in the trait, I will prefere to have a small Helper service to replace the trait. I will push something